chain/cosmos/module_auth.go

```diff
@@ -0,0 +1,153 @@
+
+// AuthGetModuleAccount performs a query to get the account details of the specified chain module
+func (c *CosmosChain) AuthQueryModuleAccount(ctx context.Context, moduleName string) (authtypes.ModuleAccount, error) {
+	res, err := authtypes.NewQueryClient(c.GetNode().GrpcConn).ModuleAccountByName(ctx, &authtypes.QueryModuleAccountByNameRequest{
+		Name: moduleName,
+	})
+	if err != nil {
+		return authtypes.ModuleAccount{}, err
+	}
+
+	var modAcc authtypes.ModuleAccount
+	err = c.GetCodec().Unmarshal(res.Account.Value, &modAcc)
+
+	return modAcc, err
+}
+
+// GetModuleAddress performs a query to get the address of the specified chain module
+func (c *CosmosChain) AuthQueryModuleAddress(ctx context.Context, moduleName string) (string, error) {
+	queryRes, err := c.AuthQueryModuleAccount(ctx, moduleName)
+	if err != nil {
+		return "", err
+	}
+	return queryRes.BaseAccount.Address, nil
+}
+
+// Deprecated: use AuthQueryModuleAddress instead
+func (c *CosmosChain) GetModuleAddress(ctx context.Context, moduleName string) (string, error) {
+	return c.AuthQueryModuleAddress(ctx, moduleName)
+}
+
+// GetGovernanceAddress performs a query to get the address of the chain's x/gov module
+// Deprecated: use AuthQueryModuleAddress(ctx, "gov") instead
+func (c *CosmosChain) GetGovernanceAddress(ctx context.Context) (string, error) {
+	return c.GetModuleAddress(ctx, "gov")
+}
+
+func (c *CosmosChain) AuthQueryBech32Prefix(ctx context.Context) (string, error) {
+	res, err := authtypes.NewQueryClient(c.GetNode().GrpcConn).Bech32Prefix(ctx, &authtypes.Bech32PrefixRequest{})
+	return res.Bech32Prefix, err
+}
+
+// AddressBytesToString converts a byte array address to a string
+func (c *CosmosChain) AuthAddressBytesToString(ctx context.Context, addrBz []byte) (string, error) {
+	res, err := authtypes.NewQueryClient(c.GetNode().GrpcConn).AddressBytesToString(ctx, &authtypes.AddressBytesToStringRequest{
+		AddressBytes: addrBz,
+	})
+	return res.AddressString, err
+}
+
+// AddressStringToBytes converts a string address to a byte array
+func (c *CosmosChain) AuthAddressStringToBytes(ctx context.Context, addr string) ([]byte, error) {
+	res, err := authtypes.NewQueryClient(c.GetNode().GrpcConn).AddressStringToBytes(ctx, &authtypes.AddressStringToBytesRequest{
+		AddressString: addr,
+	})
+	return res.AddressBytes, err
+}
+
+// AccountInfo queries the account information of the given address
+func (c *CosmosChain) AuthQueryAccountInfo(ctx context.Context, addr string) (*authtypes.BaseAccount, error) {
+	res, err := authtypes.NewQueryClient(c.GetNode().GrpcConn).AccountInfo(ctx, &authtypes.QueryAccountInfoRequest{
+		Address: addr,
+	})
+	return res.Info, err
+}
+
+func (c *CosmosChain) AuthPrintAccountInfo(chain *CosmosChain, res *cdctypes.Any) error {
+	switch res.TypeUrl {
+	case "/cosmos.auth.v1beta1.ModuleAccount":
+		var modAcc authtypes.ModuleAccount
+		err := chain.GetCodec().Unmarshal(res.Value, &modAcc)
+		fmt.Printf("modAcc: %+v\n", modAcc)
+		return err
```

Comment:
for all of these cases should the error be checked before trying to call `Printf`? if there is an error the print calls would just print an empty struct

Reply:
https://github.com/strangelove-ventures/interchaintest/pull/892/commits/7cef1a9df16f9ca52a90cebbd9fcc458561c8961